src/main/java/com/wavefront/sdk/common/Utils.java

```diff
@@ -24,39 +24,86 @@
 
   private static final ObjectMapper JSON_PARSER = new ObjectMapper();
 
+  public static String enquote(String s) {
+    if (s.contains("\"") || s.contains("'")) {
+      // for single quotes, once we are double-quoted, single quotes can exist happily inside it.
+      s = s.replaceAll("\"", "\\\\\"");
+    }
+    return "\"" + s.replaceAll("\\n", "\\\\n") + "\"";
+  }
+
+  /**
+   * @deprecated Use enquote(sanitizeName(s)) for metric name, and enquote(sanitizeKey) for source
+   * and tag keys instead.
+   */
+  @Deprecated
   public static String sanitize(String s) {
     /*
-     * Sanitize string of metric name, source and key of tags according to the rule of Wavefront proxy.
+     * Sanitize string of metric name, source and key of tags according to the rule of Wavefront
+     * proxy
      */
-    
+    return enquote(sanitizeName(s));
```

Comment:
do you need to enquote this? this seems like behavior change from before

Reply:
i removed the enquoting from the base method (`sb.append('"')`) so we do need to enquote it